lib/connections/service.go

```diff
@@ -786,13 +786,24 @@ func (s *service) CommitConfiguration(from, to config.Configuration) bool {
 
 func (s *service) checkAndSignalConnectLoopOnUpdatedDevices(from, to config.Configuration) {
 	oldDevices := from.DeviceMap()
+	dial := false
+	s.dialNowDevicesMut.Lock()
 	for _, dev := range to.Devices {
+		if dev.Paused {
+			continue
+		}
 		oldDev, ok := oldDevices[dev.DeviceID]
-		if !ok || !util.EqualStrings(oldDev.Addresses, dev.Addresses) {
-			s.scheduleDialNow()
-			break
+		if !ok || oldDev.Paused || !util.EqualStrings(oldDev.Addresses, dev.Addresses) {
+			if !ok || oldDev.Paused {
+				s.dialNowDevices[dev.DeviceID] = struct{}{}
+			}
+			dial = true
```

Comment:
This nesting is ugly and looks like a mistake... Maybe flatten it to two separate branches, a second one just for the equalstrings?